framework/src/base/MooseApp.C

```diff
@@ -877,6 +877,13 @@ MooseApp::getCheckpointFiles()
     }
   }
 
+  return checkpoint_dirs;
+}
+
+std::list<std::string>
+MooseApp::getCheckpointFiles() const
```

Comment:
With this change of behavior you will need to update all places that called getCheckpointFiles to call getCheckpointDirectories.

Reply:
getCheckpointDirectories is called inside getCheckpointFiles. Did i miss something?

Reply:
Is this function actually used anywhere?  If not we should just remove it.  Did you need it in rattlesnake - if so you could just implement its two lines in rattlesnake instead.

Reply:
It is used in framework: `moose/framework/src/actions/SetupRecoverFileBaseAction.C`. Rattlesnake is not using it.